lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -0,0 +1,314 @@
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    #
+    # == Monei gateway
+    # This class implements Monei gateway for Active Merchant. For more information about Monei
+    # gateway please go to http://www.monei.net
+    #
+    # === Setup
+    # In order to set-up the gateway you need four paramaters: sender_id, channel_id, login and pwd.
+    # Request that data to Monei.
+    class MoneiGateway < Gateway
+      self.test_url = 'https://test.monei-api.net/payment/ctpe'
+      self.live_url = 'https://monei-api.net/payment/ctpe'
+
+      self.supported_countries = ['ES']
+      self.default_currency = 'EUR'
+      self.supported_cardtypes = [:visa, :master, :maestro, :jcb, :american_express]
+
+      self.homepage_url = 'http://www.monei.net/'
+      self.display_name = 'Monei'
+
+      # Constructor
+      #
+      # options - Hash containing the gateway credentials, ALL MANDATORY
+      #           :sender_id  Sender ID
+      #           :channel_id Channel ID
+      #           :login      User login
+      #           :pwd        User password
+      #
+      def initialize(options={})
+        requires!(options, :sender_id, :channel_id, :login, :pwd)
```

Comment:
All four parameters are supplied from Monei to the merchant, who have to user them to configure the gateway. All four parameters are required and different for every merchant.